Library/Homebrew/cmd/--selfdestruct.rb

```diff
@@ -0,0 +1,93 @@
+# Original Inspiration: Stephen Benner https://github.com/SteveBenner: https://gist.github.com/SteveBenner/11254428
+# Partially Rewritten by Dominyk Tiller https://github.com/DomT4 for Homebrew's internal use.
+# THIS WILL COMPLETELY REMOVE HOMEBREW FROM YOUR SYSTEM
+
+require "fileutils"
+require "open3"
+require "rubygems"
+
+begin
+  require "trollop"
+rescue LoadError
+  onoe "You must run `sudo /System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/gem install trollop`"
+  raise
+end
+
+module Homebrew
+  def __selfdestruct
+
+@options = Trollop::options do
+  opt :quiet, "Run script with minimal user-visible output"
+  opt :verbose, "Run script with maximum user-visible output"
+  opt :dryrun, "Don't delete anything. Just show what would happen"
+end
+
+brew_location = HOMEBREW_PREFIX
+
+brewlocal = %W{.git Cellar Library/brew.rb Library/Homebrew Library/Aliases Library/Formula Library/Contributions Library/ENV Library/LinkedKegs}
+$files = []
+
+# Found Homebrew installation
+if brew_location
+  unless @options[:quiet]
+    puts "Homebrew found at: #{brew_location}"
+    begin # record kegs and taps for later output
+      brewed = `brew list`
+      tapped = `brew tap`
+    rescue StandardError
+    end
+  end
+  # Collect files indexed by git
+  begin
+    Dir.chdir(brew_location) do
+      # Update file list (use popen3 so we can suppress git error output)
+      Open3.popen3('git checkout master') { |stdin, stdout, stderr| stderr.close }
+      $files += `git ls-files`.split.map {|file| File.expand_path file }
+    end
+  rescue StandardError => e
+    puts e
+  end
+end
+
+# Collect files
+$files += brewlocal.select { |file| File.exist? file }.map {|file| File.expand_path file }
+cache = "#{HOMEBREW_CACHE}"
+logs = "#{HOMEBREW_LOGS}"
+
+abort 'Failed to locate any homebrew files!' if $files.empty?
+
+rm =
+  if @options[:dryrun]
```

Comment:
Meh. I don't like interactivity, to be honest. Lots of opportunity for things to go wrong. I'm happy to stick an interactive message back towards the top, as I had before, along the lines of ` THIS WILL DESTROY YOUR HOMEBREW. Abort unless /yes/y` or similar - But I don't really want to make it super-interactive. I think we'd be causing maintenance nightmares for ourselves with interactivity.